Sources/Datadog/RUM/RUMVitals/VitalRefreshRateReader.swift

```diff
@@ -0,0 +1,35 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import UIKit
+
+internal class VitalRefreshRateReader: VitalReader {
+    private var lastRefreshRate: Double?
+
+    init() throws {
+        guard let link = UIScreen.main.displayLink(
+            withTarget: self,
+            selector: #selector(displayTick(link:))
+        ) else {
+            throw InternalError(description: "CADisplayLink could not be created!")
+        }
+        link.add(to: .main, forMode: .default)
```

Comment:
[The doc](https://developer.apple.com/documentation/quartzcore/cadisplaylink/1621323-add) says:
> The run loop retains the display link. To remove the display link from all run loops, send an `invalidate()` message to the display link.

We need to take this into account as it might be causing flakiness (in tests) and might leak memory (in tests and in production code) depending on how this component will be used from the caller.

Will we instantiate several `VitalRefreshRateReader` instances or use a single one?

Reply:
good point 👍 made a change: added `start()/stop()` which adds `link` to run loop and invalidates it later.

i think we should have an always-running singleton (`RefreshRateReader`) and multiple observers (`[VitalObserver]`) so that the usage will be like that:
```
// RUMView starts
let newObserver = createObserverForNewRUMView(
reader.shared.subscribe(newObserver)
...
// RUMView ends
let refreshRateForView = newObserver.meanValue
reader.shared.unsubscribe(newObserver)
```
i think a singleton is a good fit here because, like in `AppStateListener` case, we hit main thread when we start `reader`. that's because [Apple](https://developer.apple.com/documentation/foundation/runloop) says `RunLoop`s cannot be called from different threads.

how does that sound to you?

currently i tried to port `RefreshRateReader` logic in https://github.com/DataDog/dd-sdk-android/pull/640/ to iOS. 
i haven't figured out yet how they read refresh rate per `RUMView`. i'm still reading Android code.

Reply:
Having single instance of "refresh rate reader" makes sense 👍. It doesn't need to be singleton - IMO it sounds reasonable to store it in `RUMFeature.instance` as vitals monitoring is purely RUM-related stuff (also it doesn't need to be instantiated if RUM is disabled).

For managing the subscription in a thread-safe way you may want to have a look at existing [ValuePublisher](https://github.com/DataDog/dd-sdk-ios/blob/master/Sources/Datadog/Core/Utils/ValuePublisher.swift). It is made for performant reads and non-blocking writes using concurrent queue.